image/image.go

```diff
@@ -49,6 +49,13 @@ var validRefMediaTypes = []string{
 }
 
 func validate(w walker, refs []string, out *log.Logger) error {
+	var tmperr error
```

Comment:
> IMHO, the regular expression check should be at the beginning of the function.

I hadn't thought about that, but not that you bring it up… ;).  I think it should be [here][1], because:

1. If you run your check too early, you might not check anything and have invalid refs [injected here][2] slip through without getting caught.
2. A ref [not existing][3] seems more important than not meeting the regexp.

I feel pretty strongly about (1), but not particularly strongly about (2).  Regardless of how (2) goes, the check should be inside that loop.  Perhaps the point that the validated refs may not be parameters (when they come from `listReferences`) is enough to allow closer integration between this regexp validation and the existing logic.

[1]: https://github.com/opencontainers/image-tools/blob/0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27/image/image.go#L77
[2]: https://github.com/opencontainers/image-tools/blob/0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27/image/image.go#L63-L65
[3]: https://github.com/opencontainers/image-tools/blob/0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27/image/image.go#L75